app/controllers/declaration/AdditionalInformationController.scala

```diff
@@ -93,15 +93,12 @@ class AdditionalInformationController @Inject()(
       .fold(
         formWithErrors => Future.successful(BadRequest(additionalInformationPage(mode: Mode, itemId, formWithErrors, cachedData))),
         updatedCache =>
-          if (updatedCache != cachedData)
-            updateCache(itemId, AdditionalInformationData(updatedCache))
-              .map(
-                _ =>
-                  navigator
-                    .continueTo(controllers.declaration.routes.DocumentsProducedController.displayPage(mode, itemId))
-              )
-          else
-            Future.successful(navigator.continueTo(controllers.declaration.routes.DocumentsProducedController.displayPage(mode, itemId)))
+          updateCache(itemId, AdditionalInformationData(updatedCache))
+            .map(
+              _ =>
+                navigator
+                  .continueTo(controllers.declaration.routes.DocumentsProducedController.displayPage(mode, itemId))
+          )
```

Comment:
This is deliberate.   If we don't update the database (even with no additional information) then we don't know that the user has visited this page.    I had to do this in a couple of places (and I think I've missed a few that have been spotted by acceptance tests).

Basically we want the db to go from `None` to `Some[Seq.empty]` (or whatever the page data is)

Reply:
This if you can find on the pages that contains multiple elements. It was because if user click just save and continue we don't want to make the same call to the database.

But yeah, I understand why here it's necessary.